packages/flutter_tools/templates/app/pubspec.yaml.tmpl

```diff
@@ -1,8 +1,9 @@
 name: {{projectName}}
 description: {{description}}
-{{#withPluginHook}}
-publish_to: 'none'
-{{/withPluginHook}}
+
+# The following line prevents the package from being accidentally published to
+# pub.dev using `pub publish`. This is preferred for private packages.
+publish_to: 'none' # Remove this line if you wish to publish to pub.dev
```

Comment:
nit - maybe just move the comment to the line above

Reply:
nit: "private packages and applications" might be better, since many of our developers wouldn't understand that their app is also a package.